Creation of secretPage function and funfact html page

J-coder13 opened the pull request.

J-coder13 committed:
Creation of secretPage function and funfact html page

saucepwn commented on portfolio/src/main/webapp/gallery.html:
```diff
@@ -1,15 +1,15 @@
   <body>
     <div id="content">
-      <h1 class="center">Pictures of Jaden</h1>
+      <h1 class="center"> Pictures of Jaden</h1>
```
remove extra space

J-coder13 replied:
Got it.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
+
+/*
```
Multiline comments should be formatted like:
```
/*
 * This is a 
 * multi line comment
 */
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+ In order to access the secret page you have to solve a randomly generated equation.
+ Upon the input of a correct answer a button will appear, linking the user to the page.
+ */
+function secretPage(){
```
nit: Google style would have the curly brace separated from the parameter list by a space:
```suggestion
function secretPage() {
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+ */
+function secretPage(){
+  //Generates a number between 1 and 200 for both numbers.
+  this.a = Math.floor((Math.random()*200)+100);
```
This line is duplicated twice and should be moved to its own method to avoid code duplication

J-coder13 replied:
Got it.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+ */
+function secretPage(){
+  //Generates a number between 1 and 200 for both numbers.
+  this.a = Math.floor((Math.random()*200)+100);
```
this.a will create an instance variable attached to the page. It looks like this variable is only used in this function, so you can declare it as a local variable:

var a = ...
var b = ...

J-coder13 replied:
Got it

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+
+  //If b is a greater number we flip the values
+  if(this.b > this.a){
+    var hold =this.b;
```
nit: Keep spacing consistent

```suggestion
    let hold = this.b;
```

J-coder13 replied:
Got it

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+  
+  //Generates which equation option we choose;
+  let choice = Math.floor((Math.random()*4));
+  
```
Avoid double blank lines

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+  let input = 0;
+  let answer = 0;
+
+//choice determines which equation the user gets 
```
Indent comments as you would code, so this should be indented +2 spaces.
```suggestion
  // choice determines which equation the user gets 
```

J-coder13 replied:
Got it

saucepwn commented on portfolio/src/main/webapp/style.css:
```diff
@@ -27,3 +27,7 @@
 }
+
+#secret-button{
+    display: none;
```
Keep indentation consistent throughout the file (2 spaces)

J-coder13 replied:
Got it

saucepwn commented on portfolio/src/main/webapp/talent.html:
```diff
@@ -0,0 +1,17 @@
+
+  <head>
+    <meta charset="utf-8">
+    <title>Pictures Of Jaden</title>
```
If this is a placeholder for the talent page, can you change the page title to identify it as such?

J-coder13 replied:
Got it.

saucepwn commented on portfolio/src/main/webapp/talent.html:
```diff
@@ -0,0 +1,17 @@
+
+  <body>
+    <div id="content">
+      <h1 class="center"> Jaden's Fun Fact</h1>
```
Remove errant space after the opening `<h1>` tag unless there's a need for it

J-coder13 replied:
Got it.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+  }
+
+  // This section of code includes the possible equations to solve
+  this.div = function(){
```
This is an interesting way of doing this. It's good to demonstrate how to save functions to a variable (called closures in Javascript) but in this context it makes the code more complex and harder to read. Instead of saving these operations to function closures, you could directly do perform the operation in the switch statement on line 72

J-coder13 replied:
Got it.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,94 @@ function addRandomGreeting() {
+      case 0:
+        answer = this.add();
+        input = prompt(this.a +' + ' + this.b + '?',0);
+        if(input == answer){ 
```
Here's a fun shorthand to collapse these three lines into one:

isCorrect = input == answer;

J-coder13 replied:
Cool I definitely implement that.

J-coder13 committed:
Renames talent.html to funfact. Improves readability of secretPage().
Messages for secretPage() improved to give the user bettr understanding
of what the button does.

J-coder13 committed:
Finishes fills funFact.html with content.

J-coder13 committed:
Change to Data servlet

J-coder13 committed:
Displays text from data servlet.

J-coder13 committed:
Update script.js

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -27,35 +27,26 @@ function addRandomGreeting() {
 }
 
+//Generates a number between 1 and 200
+function randomNumber() {
```
There are two places where random numbers are required (one between [1...4], the other [1...200]). Can you make this function take an input parameter to specify the highest number so you can use it in both cases?

J-coder13 replied:
Yeah that makes more sense.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -27,35 +27,26 @@ function addRandomGreeting() {
   //Generates a number between 1 and 200 for both numbers.
-  this.a = Math.floor((Math.random()*200)+100);
-  this.b = Math.floor((Math.random()*200)+100);
+  var a = randomNumber();
```
Instead of using `var`, which preceedes `const` and `let`, use either `const` or `let`. Since these values will be reassigned, `let` would be the right choice here.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -27,35 +27,26 @@ function addRandomGreeting() {
-  } 
-  this.mul = function(){
-    return this.a * this.b;   
+  if(b > a){
```
Try to keep styling consistent
```suggestion
  if(b > a) {
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -27,35 +27,26 @@ function addRandomGreeting() {
+ * In order to access the secret page you have to solve a randomly generated equation.
+ * Upon the input of a correct answer a button will appear, linking the user to the page.
  */
 function secretPage(){
```
Consistent styling
```suggestion
function secretPage() {
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -64,43 +55,34 @@ function secretPage(){
-          isCorrect = true;
-        }
+        answer = a + b;
+        input = prompt(a +' + ' + b + '?',0);
```
Spacing consistency
```suggestion
        input = prompt(a + ' + ' + b + '?', 0);
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -64,43 +55,34 @@ function secretPage(){
-          isCorrect = true;
-        }
+        answer = a - b;
+        input = prompt(a +' - ' + b + '?',0);
```
Spacing consistency
```suggestion
        input = prompt(a + ' - ' + b + '?', 0);
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -64,43 +55,34 @@ function secretPage(){
-          isCorrect = true;
-        }
+        answer = a * b;
+        input = prompt(a +' * ' + b + '?',0);
```
Spacing consistency
```suggestion
        input = prompt(a + ' * ' + b + '?', 0);
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -64,43 +55,34 @@ function secretPage(){
-        }
-       break;
+        answer = Math.round(a / b );
+        input = prompt(a +' / ' + b + '? Rounded to the nearest Integer',0);
```
Spacing consistency
```suggestion
        input = prompt(a + ' / ' + b + '? Rounded to the nearest Integer', 0);
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -109,11 +91,18 @@ function secretPage(){
     //The button to access the secret page is hidden until the user enters a correct answer 
     let secret = document.getElementById('secret-button');
```
`secret` can be marked `const` instead of `let` since the value of `secret` is never reassigned

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -109,11 +91,18 @@ function secretPage(){
     //The button to access the secret page is hidden until the user enters a correct answer 
     let secret = document.getElementById('secret-button');
     if(isCorrect){
```
Spacing consistency
```suggestion
    if (isCorrect) {
```

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -109,11 +91,18 @@ function secretPage(){
-        alert('Congratulations! You have unlocked my Secret.');
+        alert('Congratulations, You have passed the test! Now click the button at the bottom to see my hidden talent');
         secret.style.display = 'block';
     }else{
```
Spacing consistency
```suggestion
    } else {
```

J-coder13 committed:
Makes spacing consistent. Fixes disappearing photo problem. Optimizes randomNumber().

J-coder13 committed:
Merge branch 'jaden_development' of https://github.com/J-coder13/my-portfolio into jaden_development

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -38,13 +38,13 @@ function randomNumber() {
-  if(b > a){
-    var hold = b;
+  if (b > a) {
+    let hold = b;
```
`let` can be `const` here since the variable `hold` is not reassigned after initialization.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -53,7 +53,7 @@ function secretPage(){
   
   //Generates which equation option we choose;
-  let choice = Math.floor((Math.random()*4));
+  let choice = randomNumber(4, 0);
```
`choice` can be `const` since it is never reassigned after initialization.

saucepwn commented on portfolio/src/main/webapp/script.js:
```diff
@@ -38,13 +38,13 @@ function randomNumber() {
   //Generates a number between 1 and 200 for both numbers.
-  var a = randomNumber();
-  var b = randomNumber();
+  let a = randomNumber(200,1);
+  let b = randomNumber(200,1);
```
Per the Google Javascript style guide: https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace

> Horizontal Whitespace is required:
> After a comma (,) or semicolon (;). Note that spaces are never allowed before these characters.

```suggestion
  let a = randomNumber(200, 1);
  let b = randomNumber(200, 1);
```

saucepwn reviewed (approved):
Looking good, feel free to merge this PR right after addressing comments - no need for another review unless you have any unanswered questions.
